app/src/main/kotlin/com/flxrs/dankchat/preferences/ui/ChatSettingsFragment.kt

```diff
@@ -124,4 +146,55 @@ class ChatSettingsFragment : MaterialPreferenceFragmentCompat() {
 
         return true
     }
+
+    private fun showUserDisplaySettingsFragment(root: View): Boolean {
+        val context = root.context
+        val windowHeight = resources.displayMetrics.heightPixels
+        val peekHeight = (windowHeight * 0.6).roundToInt()
+
+        lifecycleScope.launch {
```

Comment:
Create a UserDisplayViewModel and move all of this logic into there please

Reply:
A UI component should never directly communicate with a component in the data layer, instead it should be delegated to a state holder, in this case a viewholder. See also https://developer.android.com/topic/architecture#ui-layer

Reply:
sure :+1: